### PR TITLE
pcd_viewer: Add support for absolute repositioning of point clouds.

### PR DESCRIPTION
**Pull request by j-oberlaender**
This is useful when visualizing multiple .pcd files together.

### CODE REVIEW

File: visualization/tools/pcd_viewer.cpp [hunk 1]
```diff
@@ -422,6 +428,19 @@ main (int argc, char** argv)
     if (pcd.read (argv[p_file_indices.at (i)], *cloud, origin, orientation, version) < 0)
       return (-1);
 
+    // Calculate transform if available.
+    if (pose_x.size () > i && pose_y.size () > i && pose_z.size () > i &&
+        pose_roll.size () > i && pose_pitch.size () > i && pose_yaw.size () > i)
+    {
```
Comment (mschoeler) [inline visualization/tools/pcd_viewer.cpp]: sorry I just saw this nice addition.
wouldn't it make sense if you could only give position parameters?
I think this is a very common application, when you for example want to show multiple object pcds in one run, just displaced by lets say 1 meter from each other?
For that a very nice addition would be to check if the pose_x,y,z sizes are one, i.e. only one displacement vector **d**
In that case the standard behaviour would be to displace each pointcloud with index *Ind* by **d** * *Ind* ?
What do you think?

Comment (j-oberlaender) [inline visualization/tools/pcd_viewer.cpp]: Good idea to allow just translation. I would then just check the translation and rotation vector sizes separately, and assume an identity rotation matrix where none is given. I've just pushed a commit to my clone doing that, how do you like it?

**Comment by j-oberlaender** [comment]
Not sure how best to link to that commit... it's here: https://github.com/j-oberlaender/pcl/commit/0acb61ebdc96b4eebeaf297ce09222bb23217df9

File: visualization/tools/pcd_viewer.cpp (hunk 1 above)
Comment (j-oberlaender) [inline visualization/tools/pcd_viewer.cpp]: https://github.com/j-oberlaender/pcl/commit/0acb61ebdc96b4eebeaf297ce09222bb23217df9

**Pull request merged**